### PR TITLE
samples: openthread: cert: update THCI to the latest vestion

### PR DESCRIPTION
**Pull request by MarekPorwisz**
OpenThread and it's THCI file underwent a lot of changes and currently
used by NCS THCI file is outdated and causes certification failures.
THCI in OpenThread supports also zephyr right now so this version was
copied to NCS with classes being updated properly. Without changes
included in this file Thread 1.1 certification will fail.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]
This is code duplication. Import/inherit from vanilla OT THCI and overwrite what needs to be changed to prevent from bugs in the future.

**Comment by MarekPorwisz** [comment]
> This is code duplication. Import/inherit from vanilla OT THCI and overwrite what needs to be changed to prevent from bugs in the future.
This file is different from the one in zephyrs openthread. In addition to changed class names it is based on modifications done by Kacper that allow our Thread 1.1 certification to pass. Those changes were tested to work with our NCS solution but are still not present even in Vanilla OpenThread.

File: samples/openthread/cli/harness/nRF_Connect_SDK.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env python
 #
-# Copyright (c) 2020 Nordic Semiconductor ASA
+# Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (edmont) [inline samples/openthread/cli/harness/nRF_Connect_SDK.py:3]: This change is not needed since it only states the date of the creation of the file.

**Review by edmont — APPROVED** [review]

**Comment by edmont** [comment]
> This is code duplication. Import/inherit from vanilla OT THCI and overwrite what needs to be changed to prevent from bugs in the future.

I agree this is code duplication. However the file is only provided as convenience in OpenThread, there is no upstream CI testing it, neither downstream one.

I believe the file provided by this PR is okay for the current SDK code, and we are just doing the same thing as we have done for this file in the past. In the future we should find the way to integrate the file into some CI.

**Comment by lmaciejonczyk** [comment]
> I believe the file provided by this PR is okay for the current SDK code, and we are just doing the same thing as we have done for this file in the past. In the future we should find the way to integrate the file into some CI.

I guess the right approach is to have all required changes in upstream and here just import the classes like:
```
from OpenThread import OpenThread
class nRF_Connect_SDK(OpenThread):
```
if we really needs the different class name for GRL(?)
I understand that this is too late for making these changes for 1.8 release, but it looks it's not so hard to make it fix just after. I'm granting green light for the PR.

**Review by lmaciejonczyk — APPROVED** [review]

File: samples/openthread/cli/harness/nRF_Connect_SDK.py (hunk 1 above)
Comment (MarekPorwisz) [reply samples/openthread/cli/harness/nRF_Connect_SDK.py:3]: I thought file modification resets the copyrights timeout but thanks for clarification. Will push the change in a moment.

**Pull request merged**